File: app/src/test/java/com/mgaetan89/showsrage/network/SickRageApi_GetWebRootTest.java
package com.mgaetan89.showsrage.network;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class SickRageApi_GetWebRootTest {
	@Parameterized.Parameter(0)
	public String apiPath;

	@Parameterized.Parameter(1)
	public String webRoot;

	@Test
	public void getWebRoot() {
		assertThat(SickRageApi.getWebRoot(this.apiPath)).isEqualTo(this.webRoot);
	}

	@Parameterized.Parameters(name = "{index}: {0} -> {1}")
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null, ""},
				{"", ""},
				{"api", ""},
				{"/api", ""},
				{"/api/", ""},
				{"foo/api", "foo/"},
				{"foo/api/", "foo/"},
				{"/foo/api", "foo/"},
				{"/foo/api/", "foo/"},
				{"api/bar", "api/bar/"},
				{"api/bar/", "api/bar/"},
				{"/api/bar", "api/bar/"},
				{"/api/bar/", "api/bar/"},
				{"foo/api/bar", "foo/api/bar/"},
				{"foo/api/bar/", "foo/api/bar/"},
				{"/foo/api/bar", "foo/api/bar/"},
				{"/foo/api/bar/", "foo/api/bar/"},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/network/SickRageApi_GetWebRootTest.kt
package com.mgaetan89.showsrage.network

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class SickRageApi_GetWebRootTest(val apiPath: String?, val webRoot: String) {
    @Test
    fun getWebRoot() {
        assertThat(SickRageApi.getWebRoot(this.apiPath)).isEqualTo(this.webRoot)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters(name = "{index}: {0} -> {1}")
        fun data(): Collection<Array<Any?>> {
            return listOf(
                    arrayOf<Any?>(null, ""),
                    arrayOf<Any?>("", ""),
                    arrayOf<Any?>("api", ""),
                    arrayOf<Any?>("/api", ""),
                    arrayOf<Any?>("/api/", ""),
                    arrayOf<Any?>("foo/api", "foo/"),
                    arrayOf<Any?>("foo/api/", "foo/"),
                    arrayOf<Any?>("/foo/api", "foo/"),
                    arrayOf<Any?>("/foo/api/", "foo/"),
                    arrayOf<Any?>("api/bar", "api/bar/"),
                    arrayOf<Any?>("api/bar/", "api/bar/"),
                    arrayOf<Any?>("/api/bar", "api/bar/"),
                    arrayOf<Any?>("/api/bar/", "api/bar/"),
                    arrayOf<Any?>("foo/api/bar", "foo/api/bar/"),
                    arrayOf<Any?>("foo/api/bar/", "foo/api/bar/"),
                    arrayOf<Any?>("/foo/api/bar", "foo/api/bar/"),
                    arrayOf<Any?>("/foo/api/bar/", "foo/api/bar/")
            )
        }
    }
}
